driver-core/src/main/com/mongodb/event/ConnectionCheckOutFailedEvent.java

```diff
@@ -53,19 +53,36 @@ public enum Reason {
     }
 
     private final ServerId serverId;
+    private final long operationId;
+
     private final Reason reason;
 
     /**
      * Construct an instance
      *
      * @param serverId the server id
+     * @param operationId the operation id
      * @param reason the reason the connection check out failed
+     * @since 4.10
      */
-    public ConnectionCheckOutFailedEvent(final ServerId serverId, final Reason reason) {
+    public ConnectionCheckOutFailedEvent(final ServerId serverId, final long operationId, final Reason reason) {
```

Comment:
Internally we use `OperationContext`, but destruct it when exposing its data to a user via events. The reason we use `OperationContext` internally is, I am guessing, to allow for future expansion of the data associated with an operation. But if we internally take that possibility into account, shouldn't we do the same in the API and expose a `OperationContext` to a user instead of deconstructing `OperationContext`?

If we do this, we should expose an interface with read-only methods, because users must not be able to mutate the context even if internally we'll do that in the future.

Reply:
I'm not so sure about this.  The reason I actually made OperationContext a class is not to allow for future expansion, as that can always be done as needed with a simple refactoring.  Rather, I did it for code clarity.  I typically don't like passing primitives around as it's easy to mix them up with other primitives of the same type (e.g. the request id).

In practice, operationId has been in the spec for years and I haven't heard of any need to expand it, though one never knows.

Reply:
Got it, it's what is called the [new type pattern](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) in Rust. Then it seems, `PooledConnection.operationId` should be replaced with `PooledConnection.operationContext`, shouldn't it?

Reply:
> Got it, it's what is called the [new type pattern](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) in Rust. Then it seems, PooledConnection.operationId should be replaced with PooledConnection.operationContext, shouldn't it?

Seems reasonable.  Done.